ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AlterTableHandler.java

```diff
@@ -82,14 +90,49 @@ private Scenario scenarioType(org.apache.hadoop.hive.metastore.api.Table before,
     }
   }
 
+  // return true, if event needs to be dumped, else return false.
+  private boolean handleRenameDuringTableLevelReplication(Context withinContext) {
+    String oldName = before.getTableName();
+    String newName = after.getTableName();
+
+    if (ReplUtils.tableIncludedInReplScope(withinContext.replScope, oldName)) {
+      // If old table satisfies the filter, but the new table does not, then the old table should be dropped.
+      if (!ReplUtils.tableIncludedInReplScope(withinContext.replScope, newName)) {
+        scenario = Scenario.DROP;
+        LOG.info("Table " + oldName + " will be dropped as the table is renamed to " + newName);
+        return true;
+      }
+
+      // If both old and new table satisfies the filter, then dump the rename event.
+      return true;
```

Comment:
For case, out1 -> in1 -> in2, in2 should be part of bootstrap list but it won't as caller has removed in1 from the list and here we don't add in2 to it.